cli/commands/run/version_check.go

```diff
@@ -212,6 +237,31 @@ func parseTerraformImplementationType(versionCommandOutput string) (options.Terr
 	}
 }
 
+// Helper to compute a cache key from the checksums of .terraform-version and .tool-versions
+func computeVersionFilesCacheKey(workingDir string) string {
+	var hashes []string
+
+	files := []string{".terraform-version", ".tool-versions"}
+
+	for _, file := range files {
+		path := filepath.Join(workingDir, file)
+		if util.FileExists(path) {
+			hash, err := util.FileSHA256(path)
+			if err == nil {
+				hashes = append(hashes, file+":"+hex.EncodeToString(hash))
+			}
+		}
+	}
+
+	cacheKey := "no-version-files"
+
+	if len(hashes) != 0 {
+		cacheKey = strings.Join(hashes, "|")
+	}
+
+	return util.EncodeBase64Sha1(cacheKey)
```

Comment:
NIT: Should we be SHA-ing this? It's going to get hashed by the standard library when used for the key of the map, right? I don't think it's super important, just trying to confirm my understanding right.

Reply:
`util.FileSHA256(path)` - returns 8kb byte array, encoded in hex it is quite large key.

`sha1.Sum([]byte(str))` - return 20 bytes

Reply:
I was just saying that I don't know that we need to hash this, as it's used as the key for the hashmap, which is going to get hashed by the standard library, right?